### ISSUE TITLE
Error reporting seriously lags behind

### ISSUE THREAD

**Issue by anthonyvdotbe** [oracle/javavscode#49]
The extension is very slow to realize when errors have been resolved. Doing `Java: Clean Workspace` clears the false positives, but that's not a solution (and sometimes the cleaning itself hangs and I need to restart VS Code altogether).

**Comment by lahodaj** [comment]
I think some a little bit more description of what exactly you do and see would be helpful. I don't normally see very slow error updates, but there's probably some information I am missing. One particular thing that comes to mind is whether the errors get updated if you do File/Save All. Thanks!

**Comment by anthonyvdotbe** [comment]
Here's an example of 2 warnings which persist, even after saving all files. Maybe the logs in #65 can help. They contain warnings like:

```
WARNING [org.netbeans.modules.java.hints]: document changed, but not canceled?
WARNING [org.netbeans.modules.java.hints]: len = 872
WARNING [org.netbeans.modules.java.hints]: startOffset = 872
WARNING [org.netbeans.modules.java.hints]: endOffset = 872
```

![Screenshot 2023-11-04 095815](https://github.com/oracle/javavscode/assets/15714253/da1fddd8-690e-4c68-b48a-34e78eb66e52)

**Comment by anthonyvdotbe** [comment]
An example of autocompletion lagging behind:

in file ContentNormalizer.java I've changed the second record component from `boolean dirtyBit` to `List<String> reasonsForRewriting`:

![image](https://github.com/oracle/javavscode/assets/15714253/5c66b09f-5407-4116-9fb4-00a905749ae3)

However, when attempting to adapt code in other classes, autocompletion still only gives the old component:

![image](https://github.com/oracle/javavscode/assets/15714253/ef5c6e08-2a4b-48a5-ae6b-0204955961b0)

**Comment by anthonyvdotbe** [comment]
Another example where adding a method isn't noticed elsewhere in the code (even after having saved all files and running a Maven build outside VS Code to verify this is indeed a false positive):

![image](https://github.com/oracle/javavscode/assets/15714253/b057f80b-195f-407e-a7bd-3bb76e5e9006)

**Comment by anthonyvdotbe** [comment]
Another example. I just opened VS Code, opened the file, deleted a block of code within a method which had no impact on the correctness of the method (verified with a successful `mvn clean verify` outside VS Code), yet suddenly it gives me the error below:

![image](https://github.com/oracle/javavscode/assets/15714253/6a25c072-9c87-409e-bab8-89d143589ddf)

where links is defined as:

```java
var links = new ArrayList<String>();
```

Note how the first line correctly says for `join(String,List<String>)` (for which there *are* suitable methods), but then during the actual matching it seems to think there's a variable of type `Path` involved.

**Comment by anthonyvdotbe** [comment]
This [NetBeans issue](https://bz.apache.org/netbeans/show_bug.cgi?id=253276) which I filed at the time was very similar.

For reproducing, making a change like adding a parameter to a method, renaming a method, removing an enum constant (all without using a refactoring) should do the trick. I just tried:
* renaming an enum constant: a usage of the constant elsewhere (in a class in another package) was not detected as an error
* adding a parameter to a method: invocations of the method elsewhere (in a class in another package) were not detected as an error (navigating to the method from the other class, opens the file that contains the method at the beginning of the file)

Doing `Java: Clean Workspace` *does* mark the errors due to the changes.
